### PR TITLE
Post endpoint sell assets

### PR DESCRIPTION
**Pull request by Hypermood**

### CODE REVIEW

File: src/main/java/com/melon/portfoliomanager/controllers/AssetController.java [hunk 1]
```diff
@@ -42,4 +43,21 @@ public ResponseEntity<String> buyAssets(@Valid @RequestBody TransactionDto trans
...
+
+        try {
+            assetService.sellStock(transactionDto);
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+        } catch (NoSuchUserException e) {
+            return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
+        } catch (NotEnoughStocksToSell e) {
+            return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
+        } catch (Exception e) {
+            logger.error(String.format("Unexpected error occurred during creating user ! exception=%s", e));
```
Comment (VladimirNachev) [inline src/main/java/com/melon/portfoliomanager/controllers/AssetController.java]: "... during creating user..." - this should be changed I think.

File: src/main/java/com/melon/portfoliomanager/services/AssetService.java [hunk 2]
```diff
@@ -71,5 +73,32 @@ public User validateUser(TransactionDto transactionDto) {
         return userList.get(0);
     }
 
+    public void sellStock(TransactionDto transactionDto) {
+
+        User user = validateUser(transactionDto);
+
+        List<PortfolioItem> portfolioItemList = portfolioItemRepository.findByUserIdAndCompanyName(user.getId(), transactionDto.getAssetSymbol());
+
+        PortfolioItem portfolioItem;
```
Comment (VladimirNachev) [inline src/main/java/com/melon/portfoliomanager/services/AssetService.java]: [nit] You can declare and initialize the ```portfolioItem``` variable on line 88.

Comment (VladimirNachev) [inline src/main/java/com/melon/portfoliomanager/services/AssetService.java]: [nit] You can declare and initialize the ```portfolioItem``` variable on line 88.

File: src/main/java/com/melon/portfoliomanager/services/AssetService.java [hunk 3]
```diff
@@ -71,5 +73,32 @@ public User validateUser(TransactionDto transactionDto) {
...
+        portfolioItem = portfolioItemList.get(0);
+
+        if (portfolioItem.getQuantity() < transactionDto.getQuantity()) {
+            throw new NotEnoughStocksToSell();
+        }
+
+        if (Objects.equals(portfolioItem.getQuantity(), transactionDto.getQuantity())) {
+            portfolioItemRepository.delete(portfolioItem);
+        } else {
+            portfolioItem.setQuantity(portfolioItem.getQuantity() - transactionDto.getQuantity());
```
Comment (VladimirNachev) [inline src/main/java/com/melon/portfoliomanager/services/AssetService.java:96]: In the future, may be we should add additional column to the ```PortfolioItem``` database table for storing something like ```totalSoldPrice``` for easier calculation of some of the metrics for the analytics endpoint.
And then, add the sell price to the ```totalSoldPrice```.

File: .idea/compiler.xml
Comment (VladimirNachev) [inline .idea/compiler.xml:1]: Is this file really needed to be stored into the repository ? Should not it be included in .gitignore file ?

File: src/main/java/com/melon/portfoliomanager/models/User.java [hunk 4]
```diff
@@ -38,6 +38,14 @@ public User(String username, String email, String firstName, String lastName) {
         this.lastName = lastName;
     }
 
+    public User(Long id, String username, String email, String firstName, String lastName) {
```
Comment (VladimirNachev) [inline src/main/java/com/melon/portfoliomanager/models/User.java]: Why do you need this constructor ? Do you use it only inside tests ?

File: src/main/java/com/melon/portfoliomanager/controllers/AssetController.java (hunk 1 above)
Comment (Hypermood) [reply src/main/java/com/melon/portfoliomanager/controllers/AssetController.java]: fixed

File: src/main/java/com/melon/portfoliomanager/services/AssetService.java (hunk 2 above)
Comment (Hypermood) [reply src/main/java/com/melon/portfoliomanager/services/AssetService.java]: fixed

Comment (Hypermood) [reply src/main/java/com/melon/portfoliomanager/services/AssetService.java]: fixed

File: src/main/java/com/melon/portfoliomanager/services/AssetService.java (hunk 3 above)
Comment (Hypermood) [reply src/main/java/com/melon/portfoliomanager/services/AssetService.java:96]: fixed

File: .idea/compiler.xml
Comment (Hypermood) [reply .idea/compiler.xml:1]: fixed

File: src/main/java/com/melon/portfoliomanager/models/User.java (hunk 4 above)
Comment (Hypermood) [reply src/main/java/com/melon/portfoliomanager/models/User.java]: fixed

**Review by VladimirNachev — APPROVED** [review]
